source/internal.d.ts

```diff
@@ -76,6 +77,16 @@ Matches non-recursive types.
 */
 export type NonRecursiveType = BuiltIns | Function | (new (...args: any[]) => unknown);
 
+/**
+Returns a boolean for whether the given type is a normal key-value object.
+*/
+export type IsRecordObject<T> =
```

Comment:
`Record` is a type, so the expectation is that this would detect such type, which I don't think it does exactly. Maybe a better name would be `IsPlainObject`?